<HTML>
<HEAD>
<TITLE>System.OverflowException Class</TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF">
<H3>System.OverflowException Class</H3>

<BLOCKQUOTE>
<TABLE COLS="1" ROWS="1" WIDTH="100%">
<TR><TD BGCOLOR="#EEEEFF"><PRE>public class OverflowException : ArithmeticException</PRE></TD></TR>
</TABLE>
</BLOCKQUOTE>

<H4>Base Types</H4>

<BLOCKQUOTE>
<A HREF="Object.html" TARGET="contents">Object</A><BR>
&nbsp;&nbsp;<A HREF="Exception.html" TARGET="contents">Exception</A><BR>
&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="SystemException.html" TARGET="contents">SystemException</A><BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<A HREF="ArithmeticException.html" TARGET="contents">ArithmeticException</A><BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;OverflowException<P>

</BLOCKQUOTE>

<H4>Assembly</H4>

<BLOCKQUOTE>
mscorlib
</BLOCKQUOTE>

<H4>Summary</H4>

<BLOCKQUOTE>
Represents the error that occurs when the result of an arithmetic operation is too large to be
      represented by the destination type.<P>

</BLOCKQUOTE>

<H4>Description</H4>

<BLOCKQUOTE>
In languages that detect overflow, this is the exception
      that gets thrown. For example, in C#, the <CODE>checked</CODE>
 keyword is
      used to detect overflow conditions. A <A HREF="OverflowException.html" TARGET="contents">OverflowException</A> exception occurs only in a
   <CODE>checked</CODE>

   context.<P>

[<I>Note</I>: The following IL instructions throw <A HREF="OverflowException.html" TARGET="contents">OverflowException</A> :<P>

<UL>
<LI>
      add.ovf.&lt;signed&gt;</LI>
<LI>
      conv.ovf.&lt;to type&gt;</LI>
<LI>conv.ovf.&lt;to type&gt;.un</LI>
<LI>
      mul.ovf.&lt;type&gt;</LI>
<LI>
      sub.ovf.&lt;type&gt;</LI>
<LI>newarr</LI>
</UL>
]<P>

</BLOCKQUOTE>

<H4>Example</H4>

<BLOCKQUOTE>
The following example demonstrates an error that causes
      a <A HREF="OverflowException.html" TARGET="contents">OverflowException</A> exception.<P>

<PRE>using System;
public class OverflowExample {
   public static void Main() {
      int i = 400;
      byte b = 0;
      try {
         checked { b = (byte)( i ); }
      }
      catch ( OverflowException e ) {
         Console.WriteLine( &quot;Error caught: {0}&quot;, e );
      }
   }
}
   </PRE>
The output is<P>

<PRE>
Error caught: System.OverflowException: Arithmetic operation resulted in an overflow.
   at OverflowExample.Main()
 </PRE>
</BLOCKQUOTE>

<H4>See Also</H4>

<BLOCKQUOTE>
<A HREF="System.html" TARGET="members">System Namespace</A>
</BLOCKQUOTE>

<H4>Members</H4>

<BLOCKQUOTE>
<P>

OverflowException Constructors<P>

<A HREF="#OverflowException() Constructor" TARGET="contents">OverflowException() Constructor</A><BR>
<A HREF="#OverflowException(System.String) Constructor" TARGET="contents">OverflowException(System.String) Constructor</A><BR>
<A HREF="#OverflowException(System.String, System.Exception) Constructor" TARGET="contents">OverflowException(System.String, System.Exception) Constructor</A><BR>
</BLOCKQUOTE>

<HR>

<A NAME="OverflowException() Constructor"><H3>OverflowException() Constructor</H3>

<BLOCKQUOTE>
<TABLE COLS="1" ROWS="1" WIDTH="100%">
<TR><TD BGCOLOR="#EEEEFF"><PRE>public OverflowException();</PRE></TD></TR>
</TABLE>
</BLOCKQUOTE>

<H4>Summary</H4>

<BLOCKQUOTE>
Constructs and initializes a new instance of the <A HREF="OverflowException.html" TARGET="contents">OverflowException</A> class.<P>

</BLOCKQUOTE>

<H4>Description</H4>

<BLOCKQUOTE>
This constructor initializes the System.OverflowException.Message property of the new instance to a system-supplied
   message that describes the error. This message takes into account the current
   system culture.<P>

The System.OverflowException.InnerException property is initialized to <CODE>null</CODE>
.<P>

</BLOCKQUOTE>

<H4>See Also</H4>

<BLOCKQUOTE>
<A HREF="OverflowException.html" TARGET="contents">System.OverflowException Class</A>, <A HREF="System.html" TARGET="members">System Namespace</A>
</BLOCKQUOTE>

</A><HR>

<A NAME="OverflowException(System.String) Constructor"><H3>OverflowException(System.String) Constructor</H3>

<BLOCKQUOTE>
<TABLE COLS="1" ROWS="1" WIDTH="100%">
<TR><TD BGCOLOR="#EEEEFF"><PRE>public OverflowException(string message);</PRE></TD></TR>
</TABLE>
</BLOCKQUOTE>

<H4>Summary</H4>

<BLOCKQUOTE>
Constructs and initializes a new instance of the <A HREF="OverflowException.html" TARGET="contents">OverflowException</A>
class.<P>

</BLOCKQUOTE>

<H4>Parameters</H4>

<BLOCKQUOTE>
<DL>
<DT>message</DT>
<DD>A <A HREF="String.html" TARGET="contents">String</A> that describes the error. The content of <I>message</I> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture.</DD>
</DL>
</BLOCKQUOTE>

<H4>Description</H4>

<BLOCKQUOTE>
This constructor initializes the System.OverflowException.Message property of the new instance using <I>message</I>. If <I>message</I> is 
<CODE>null</CODE>
, the System.OverflowException.Message property is initialized to the 
   system-supplied message provided by the constructor that takes no arguments. The System.OverflowException.InnerException property is initialized to <CODE>null</CODE>
.<P>

</BLOCKQUOTE>

<H4>See Also</H4>

<BLOCKQUOTE>
<A HREF="OverflowException.html" TARGET="contents">System.OverflowException Class</A>, <A HREF="System.html" TARGET="members">System Namespace</A>
</BLOCKQUOTE>

</A><HR>

<A NAME="OverflowException(System.String, System.Exception) Constructor"><H3>OverflowException(System.String, System.Exception) Constructor</H3>

<BLOCKQUOTE>
<TABLE COLS="1" ROWS="1" WIDTH="100%">
<TR><TD BGCOLOR="#EEEEFF"><PRE>public OverflowException(string message, Exception innerException);</PRE></TD></TR>
</TABLE>
</BLOCKQUOTE>

<H4>Summary</H4>

<BLOCKQUOTE>
Constructs and initializes a new instance of the <A HREF="OverflowException.html" TARGET="contents">OverflowException</A>
class.<P>

</BLOCKQUOTE>

<H4>Parameters</H4>

<BLOCKQUOTE>
<DL>
<DT>message</DT>
<DD>A <A HREF="String.html" TARGET="contents">String</A> that describes the error. The content of <I>message</I> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture.</DD>
<DT>innerException</DT>
<DD>An instance of <A HREF="Exception.html" TARGET="contents">Exception</A> that is the cause of the current Exception. If <I>innerException</I> is non-null, then the current Exception was raised in a catch block handling <I>innerException</I> .</DD>
</DL>
</BLOCKQUOTE>

<H4>Description</H4>

<BLOCKQUOTE>
This constructor initializes the System.OverflowException.Message property of the new instance using <I>message</I>, and
   the System.OverflowException.InnerException property using <I>innerException</I> . If <I>message</I> is
<CODE>null</CODE>
, the System.OverflowException.Message property is initialized to the 
   system-supplied message provided by the constructor that takes no arguments.<P>

[<I>Note</I>: For information on
   inner exceptions, see System.Exception.InnerException
   .]<P>

<P>

</BLOCKQUOTE>

<H4>See Also</H4>

<BLOCKQUOTE>
<A HREF="OverflowException.html" TARGET="contents">System.OverflowException Class</A>, <A HREF="System.html" TARGET="members">System Namespace</A>
</BLOCKQUOTE>

</A></BODY>
</HTML>
